src/gtc/common.py

```diff
@@ -572,6 +572,51 @@ def apply(cls, node: Node, *, symtable: Dict[str, Any]) -> List[str]:
     return root_validator(allow_reuse=True, skip_on_failure=True)(_impl)
 
 
+def validate_lvalue_dims() -> RootValidatorType:
+    """Validate lvalue dimensions using the root node symbol table."""
+
+    class _LvalueDimsValidator(NodeVisitor):
+        def __init__(self, root_symtable: Dict[str, Any]):
+            self.root_symtable = root_symtable
+
+        def visit_Node(self, node: Node, **kwargs: Any) -> None:
+            symtable = kwargs.get("symtable", self.root_symtable)
+            if hasattr(node, "loop_order"):
+                kwargs["loop_order"] = node.loop_order
+            if isinstance(node, SymbolTableTrait):
+                kwargs["symtable"] = {**symtable, **node.symtable_}
+            self.generic_visit(node, **kwargs)
+
+        def visit_AssignStmt(
+            self, node: AssignStmt, *, loop_order: LoopOrder, **kwargs: Any
+        ) -> None:
+            symtable = kwargs.get("symtable", self.root_symtable)
```

Comment:
Let's pass it only via kwargs. See suggestions.

Reply:
This is at least the pattern that I followed so far.

Reply:
No reason to deviate from the established pattern. I adjusted.